RevenueCatUI/View+PresentPaywall.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  View+PresentPaywall.swift
+//  
+//
+//  Created by Nacho Soto on 7/24/23.
+//
+
+import RevenueCat
+import SwiftUI
+
+@available(iOS 16.0, macOS 13.0, tvOS 16.0, *)
+extension View {
+
+    /// Presents a ``PaywallView`` if the given entitlement identifier is not active
+    /// in the current environment for the current `CustomerInfo`.
+    /// ```swift
+    /// var body: some View {
+    ///    YourApp()
+    ///      .presentPaywallIfNecessary(requiredEntitlementIdentifier: "pro")
+    /// }
+    /// ```
+    /// - Note: If loading the `CustomerInfo` fails (for example, if Internet is offline),
+    /// the paywall won't be displayed.
+    public func presentPaywallIfNecessary(
+        mode: PaywallViewMode = .default,
+        requiredEntitlementIdentifier: String
+    ) -> some View {
+        return self.presentPaywallIfNecessary { info in
+            !info.entitlements
+                .activeInCurrentEnvironment
```

Comment:
Can we make this `.activeInCurrentEnvironment`  a parameter here? I know some people don't like using this. I think its good to have it as the default but I think I'd like this to be something the user can change 🤷‍♂️

Reply:
That's why I added the other overload (that takes a `CustomerInfo -> Bool` closure), so you can customize it.
I thought about the parameter to change that but I think it's more awkward. I documented that it has to be active in the _current_ environment in the docstring for this one.
```swift
.presentPaywallIfNecessary(entitlement: "", inCurrentEnvironment: true)
```

It's kinda confusing if you don't fully understand it IMO, and I think it's one of those cases where making the behavior depend on a `Bool` parameter is not a great interface.

What do you think?